src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,67 @@
+package duke.command;
+
+import duke.data.TaskList;
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+public abstract class AddCommand extends Command {
+    public static String COMMAND_WORD;
+    protected String taskName;
+
+    public AddCommand() {
+        super();
+    }
+
+    public abstract void checkAndSetParameters(String parameterInput) throws DukeException;
+
+    protected abstract void checkParameters(String parameterInput) throws DukeException;
+
+    protected abstract void setParameters(String parameterInput);
+
+    public abstract void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException;
+
+    protected boolean containsTaskDescription(String parameterInput, String separator) {
+        // Trim description from parameterInput
+        String[] splits = parameterInput.split(separator, 2);
+        String description = splits[0];
+
+        // Check if description is empty or only contain whitespaces
+        description = description.replaceAll("\\s+", "");
+        if (description.equals("")) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    protected boolean containsTaskTime(String parameterInput, String separator) {
+        // Trim description from parameterInput
+        String[] splits = parameterInput.split(separator, 2);
+        String time = splits[1];
+
+        // Check if description is empty or only contain whitespaces
+        time = time.replaceAll("\\s+", "");
+        if (time.equals("")) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    protected boolean containsTimeSeparator(String parameterInput, String separator) {
+        if (!parameterInput.contains(separator)) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    protected static String[] splitTaskName(String regex, String parameterInput) {
+        String[] splits = parameterInput.split(regex, 2);
+        if (splits.length == 1) {
+            return new String[]{splits[0], ""};
+        }
```

Comment:
You might want to add a comment here explaining what's going on. It's not immediately obvious. (And it might be more appropriate to throw an Exception on invalid input rather than to return dummy data.)

Reply:
Throwing an Exception is indeed far better, thanks for pointing that out!

New version:
```
// Triggers format error when task time cannot be found
if (splits.length == 1) {
    throw new InvalidDateTimeFormatException();
}
```